File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch06/C/P06-Bisect.cpp
// Real roots of a real function by the bisection method
#include <stdio.h>
#include <math.h>
#include "roots.h"

double func(double x)
{
   return (x*x*x*x - 5e0*x*x + 4e0) * sin(2e0*x);
}

int main()
{
   double a, b, h, x, xmin, xmax;
   int ierr;

   xmin = -3.5e0;                             // limits of root search domain
   xmax =  3.5e0;
   h = 0.1e0;                           // width of root separation intervals

   a = xmin;
   while (a < xmax) {                                 // root separation loop
      b = a + h;                                     // search interval [a,b]
      ierr = Bisect(func,a,b,x);
      if ((ierr == 0) && (x != b))
         printf("x = %8.5f in (%6.2f,%6.2f)  f(x) = %7.0e\n",x,a,b,func(x));
      a = b;                                           // shift left boundary
   }
}

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch06/Python/P06-Bisect.py
# Real roots of a real function by the bisection method
from math import *
from roots import *

def func(x):
   return (x*x*x*x - 5e0*x*x + 4e0) * sin(2e0*x)

# main

xmin = -3.5e0                                  # limits of root search domain
xmax =  3.5e0
h = 0.1e0                                # width of root separation intervals

a = xmin
while (a < xmax):                                      # root separation loop
   b = a + h                                          # search interval [a,b]
   (x,ierr) = Bisect(func,a,b)
   if ((ierr == 0) and (x != b)):
      print('x = {0:8.5f} in ({1:6.2f},{2:6.2f})  f(x) = {3:7.0e}'.
            format(x,a,b,func(x)))
   a = b                                                # shift left boundary
